mango/shared/decorators.py

```diff
@@ -39,3 +41,33 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator
+
+
+def pydantic_validation(**validators):
```

Comment:
Example of usage? I dont get how i can use it. In my mind you would do something like this but doesnt seem to be what it does:

```python
from pydantic import BaseModel
from mango.shared import pydantic_validation


class Dummy(BaseModel):
	a: int
	b: str
	c: float
	

@pydantic_validation(Dummy)
def dummy_function(a:int,b:str,c:float):
	pass
	
# Should raise errors
dummy_function(a="1",b=1,c=1)
```

However, as it works now the arguments of dummy function must be a dictionary and then one validator for each argument:

```python
@pydantic_validation(first_arg=Dummy)
def dummy_function(first_arg: dict):
	pass
	
# No error
dummy_function(first_arg={"a":1,"b":"1","c":1.0})
# Error (a is a string)
dummy_function(first_arg={"a":"1","b":"1","c":1.0})
```

I think can be abstracted all cases by changing how the validators are handled

Reply:
In my initial approach I thought that having only valdiators for the lists or dicts that where going to be passed as arguments would be enough, that validating a simple string or float would be too much, but it can be changed and adapted to work as you say.

Reply:
OK for this release we can keep it like this I will add it as an issue on repo